### PR TITLE
Match more ntsc-1.2 overlay draw functions

### PR DESCRIPTION
**Pull request by cadmic**
z_eff_ss_ice_piece.c fixed by changing the signature of func_800342EC or func_8003435C (don't remember which, but I changed them both for consistency)

Thanks to @tom-overton and @inspectredc for help!

### CODE REVIEW

File: src/overlays/actors/ovl_Bg_Hidan_Sekizou/z_bg_hidan_sekizou.c [hunk 1]
```diff
@@ -312,14 +312,15 @@ Gfx* func_8088D9F4(PlayState* play, BgHidanSekizou* this, s16 arg2, MtxF* arg3,
...
+    mtx2 = mtx;
+    mtx2->xx = mtx2->yy = mtx2->zz = (0.7f * phi_f12) + 0.5f;
 
-    temp_f2 = (arg3->xx * 10.0f * phi_f12) + 20.0f;
-    arg3->xw = (temp_f2 * arg4) + this->dyna.actor.world.pos.x;
-    arg3->yw = this->dyna.actor.world.pos.y + 30.0f + (.7f * phi_f12);
-    arg3->zw = (temp_f2 * arg5) + this->dyna.actor.world.pos.z;
+    temp_f2 = ((mtx2->xx * 10.0f) * phi_f12) + 20.0f;
+    mtx2->xw = (temp_f2 * arg4) + this->dyna.actor.world.pos.x;
+    mtx2->yw = (this->dyna.actor.world.pos.y + 30.0f) + (.7f * phi_f12);
```
Comment (Dragorn421) [inline src/overlays/actors/ovl_Bg_Hidan_Sekizou/z_bg_hidan_sekizou.c:320]: these added parens are needed?

File: src/overlays/actors/ovl_Bg_Toki_Hikari/z_bg_toki_hikari.c [hunk 2]
```diff
@@ -130,8 +128,9 @@ void func_808BA274(BgTokiHikari* this, PlayState* play) {
     }
 }
 
-void func_808BA2CC(BgTokiHikari* this, PlayState* play) {
-    s32 pad[2];
+void func_808BA2CC(Actor* thisx, PlayState* play) {
+    BgTokiHikari* this = (BgTokiHikari*)thisx;
+    PlayState* play2 = (PlayState*)play;
```
Comment (Dragorn421) [inline src/overlays/actors/ovl_Bg_Toki_Hikari/z_bg_toki_hikari.c:133]: is play2 out of habit or required

File: src/overlays/actors/ovl_Boss_Ganon/z_boss_ganon.c [hunk 3]
```diff
@@ -4805,17 +4805,20 @@ static void* sLightningTextures[] = {
...
-    0,   0,   0,   255, 255, 0,   240, 231, 23,  226, 208, 46,  212, 185, 69,  198, 162, 92,
-    184, 139, 115, 170, 115, 139, 156, 92,  162, 142, 69,  185, 128, 46,  208, 114, 23,  231,
-    100, 0,   255, 0,   0,   0,   0,   0,   0,   0,   0,   0,   0,   0,   0,   0,
+static u8 sLightningEnvColors[13][3] = {
+    { 0, 0, 0 },      { 255, 255, 0 },   { 240, 231, 23 },  { 226, 208, 46 }, { 212, 185, 69 },
+    { 198, 162, 92 }, { 184, 139, 115 }, { 170, 115, 139 }, { 156, 92, 162 }, { 142, 69, 185 },
+    { 128, 46, 208 }, { 114, 23, 231 },  { 100, 0, 255 },
 };
 
+static u8 sUnknown[0xC] = { 0 };
```
Comment (Dragorn421) [inline src/overlays/actors/ovl_Boss_Ganon/z_boss_ganon.c:4820]:
```suggestion
static s32 sUnknown[3] = { 0 };
```
we prefer s32 for typing padding stuff when possible

File: src/overlays/actors/ovl_Bg_Toki_Hikari/z_bg_toki_hikari.c (hunk 2 above)
Comment (cadmic) [reply src/overlays/actors/ovl_Bg_Toki_Hikari/z_bg_toki_hikari.c:133]: uh habit. Don't think it matters much so I'll remove it

File: src/overlays/actors/ovl_Bg_Hidan_Sekizou/z_bg_hidan_sekizou.c (hunk 1 above)
Comment (cadmic) [reply src/overlays/actors/ovl_Bg_Hidan_Sekizou/z_bg_hidan_sekizou.c:320]: ah yeah this is a vestige of other head-bashing

**Review by Dragorn421 — APPROVED** [review]

**Review by fig02 — APPROVED** [review]

**Pull request merged**